analyzer/codechecker_analyzer/analyzers/cppcheck/config_handler.py

```diff
@@ -38,6 +38,12 @@ def initialize_checkers(self,
         # all the possible checkers. All the checkers that are in the default
         # profile (or configured othewise, eg.: from the cli) should be
         # already enabled at this point.
-        for checker_name, data in self.checks().items():
-            if data[0] == CheckerState.default:
-                self.set_checker_enabled(checker_name, enabled=False)
+        # This happens in two phases in order to avoid iterator invalidation.
+        # (self.set_checker_enabled() removes elements, so we can't use it
+        # while iterating over the checker list.)
+        default_state_checkers = [
+            checker_name for checker_name, data in
+            self.checks().items() if data[0] == CheckerState.default]
```

Comment:
Hiding self.__available_checkers is kinda risky in this situation. Can we somehow avoid the iterator invalidation?

Reply:
I feel this can bite us in the future.

Reply:
This comment is not too clear for me. Could you please elaborate a little?